### PR TITLE
(maint) update changelog and bump project version for 2.4.0

### PR DESCRIPTION
**Pull request by holguinj**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## 2.4.0
+
+This is a minor feature release.
+
+Features:
+* add a `utf8-string->sha256` function, directly analogous to `utf8-string->sha1`
+* add a `file->sha256` function, equivalent to reading a file's contents as a
+  UTF-8 string and hashing the result. Uses an InputStream internally to avoid
+  reading the entire file into memory at once.
```
Comment (aperiodic) [inline CHANGELOG.md]: Could you add something here about increasing the range of port numbers returned by `open-port-num` to the entire ephemeral port range of 49152 through 65535?

Comment (holguinj) [inline CHANGELOG.md]: Sure. Is that considered a bug fix?

Comment (aperiodic) [inline CHANGELOG.md]: I'm not sure, as there wasn't previously any guarantee about the possible return values of `open-port-num`. I would call it an improvement.

Comment (holguinj) [inline CHANGELOG.md]: @aperiodic updated. is that what you had in mind?

Comment (aperiodic) [inline CHANGELOG.md]: Looks good!

**Pull request merged**